### ISSUE TITLE
Disable shift key?

### ISSUE THREAD

**Issue by joebinne** [Hokuten85/XICamera#4]
Hi! I love your addon! However, every time I press shift key, it zooms the camera out just a bit and when I release it it goes back to where it was. This makes the view slowly creep up to a top view of your character. Is there any way we can disable that function?

Thanks! <3

**Comment by Hokuten85** [comment]
Windower 4?

**Comment by joebinne** [comment]
Ashita v3

**Comment by Hokuten85** [comment]
@chameco @joebinne 

Well... I think I've found out how to recreate it. It has something to do with Compact 2 Keyboard I think. Shift is apparently a strafe key. Can either of you confirm that is what you are current set to?

How to fix it... not sure yet >_<

**Comment by Hokuten85** [comment]
Windower 4 has a different implementation due to not being able to manipulate memory in lua. I suspect when strafing the game directs the camera location calculation down a different logical path and bypasses what I'm doing to the camera in the Win4 version.

**Comment by chameco** [comment]
Oh I see, I hadn't realized that the "business" of the Windower 4 implementation was in a plugin, or that it was doing something different than the code injection in the Ashita version. In that case: do you think it would be fruitful to pursue an Ashita 4 plugin version that uses the shared camera code in `XICamera.Core`? I've been looking for an excuse to hack around with the Ashita SDK, so if you're not aware of any huge technical blocker I'd be happy to write such a thing.

**Comment by chameco** [comment]
Also of note: the camera bugs out in Ashita 4 when I start casting a spell, zooming back to the game's default distance. The only way I've found to resolve this is to reload the addon. This seems to occur on both keyboard layouts, so maybe this is an unrelated issue.

**Comment by Hokuten85** [comment]
I cleaned up the repository a bit ago to get it to a place were people should be able to build the plugin versions. Finally got around to looking at this bug myself in preparation of the Horizon opening and I think I found a different "more native" method of adjusting the camera. Found some memory locations that control the min and max camera values that the game itself uses to calculate where to position the camera. I'm hoping that it behaves less buggy.  The shift strafing thing appears to be much better. I seems to behave much more closely to how the native game functions during the strafe.

https://github.com/Hokuten85/XICamera/releases/tag/v0.7

I've added a prerelease for it so people could help with some testing and give feedback.

**Comment by Lotharl** [comment]
Any plans to get this on Windower 4?

**Comment by Hokuten85** [comment]
Yup. Got the new approach working for Windower 5 last night and started in on reworking Windower 4. Hope to have it done over the next few nights.

**Comment by Lotharl** [comment]
Awesome! Thanks for the making Vana'diel a better place!

**Comment by Hokuten85** [comment]
> Awesome! Thanks for the making Vana'diel a better place!

Windower 4 version is available in the release posted above. Hopefully it works, lol.

The shift issues should be resolved at least
